frontend/src/components/common/ChangePassword/ChangePasswordModal.tsx

```diff
@@ -0,0 +1,264 @@
+import {
+  Button,
+  FormControl,
+  FormLabel,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from "@chakra-ui/react";
+import { FirebaseError } from "firebase/app";
+import {
+  getAuth,
+  signInWithEmailAndPassword,
+  updatePassword,
+} from "firebase/auth";
+import React, { useContext } from "react";
+
+import AuthContext from "../../../contexts/AuthContext";
+import firebaseApp from "../../../utils/Firebase";
+import PasswordInputField from "../PasswordInputField";
+import ChangePasswordErrorModal from "./ChangePasswordErrorModal";
+import ChangePasswordSuccessModal from "./ChangePasswordSuccessModal";
+
+/**
+ * The model defining the props for the Change Password Modal component
+ */
+interface ChangePasswordModalProps {
+  /** A boolean representing whether the modal is open or not */
+  isOpen: boolean;
+  /** A function that sets isOpen to false */
+  onClose: () => void;
+}
+
+/**
+ * Type for error-related props to the PasswordInputField component
+ */
+type PasswordFeedback = {
```

Comment:
Out of curiosity, is there a reason why we went with `type` rather than `interface`?

Reply:
no particular reason - I changed it to interface